crates/jstz_cli/src/sandbox/daemon.rs

```diff
@@ -131,32 +140,96 @@ impl Sandbox {
         if let Some(mut n) = self.octez_node.take() {
             n.shutdown()?;
         }
+        if let Some(j) = self.jstz_node.take() {
+            j.abort();
+        }
+
         self.remove_sandbox_from_config()?;
         Ok(())
     }
 
-    fn remove_sandbox_from_config(&mut self) -> Result<()> {
-        let mut config = self.config.borrow_mut();
-        config.reload()?;
-        config.sandbox = None;
-        config.save()
-    }
-
-    pub fn join(&mut self) -> Result<()> {
-        let mut threads: Vec<&mut OctezThread> = Vec::new();
-        if let Some(n) = &mut self.octez_node {
+    pub async fn join(mut self) -> Result<()> {
+        let mut threads: Vec<OctezThread> = Vec::new();
+        let mut jstz_node = Option::None;
+        if let Some(n) = self.octez_node.take() {
             threads.push(n);
         }
 
-        if let Some(b) = &mut self.octez_baker {
+        if let Some(jn) = self.jstz_node.take() {
+            jstz_node = Some(jn);
+        }
+
+        if let Some(b) = self.octez_baker.take() {
             threads.push(b)
         }
 
-        if let Some(r) = &mut self.rollup_node {
+        if let Some(r) = self.rollup_node.take() {
             threads.push(r)
         }
 
-        OctezThread::join(&mut threads)
+        let mut signals = Signals::new([SIGINT, SIGTERM])?;
+
+        // Loop to notify when either the octez binaries or jstz node is not running
+        // or if SIGINT/SIGTERM was received. This ensures that all processes are
+        // shutdown when any process is shutdown.
+        self.task_set
+            .run_until(async move {
+                task::spawn_local(async move {
+                    'main_loop: loop {
+                        for thread in threads.iter() {
+                            if !thread.is_running() {
+                                println!(
```

Comment:
We tend to prefer the `info!` macro over `println!` (although it isn't correctly configured to log to stdout atm, but stderr instead 😔 )

Reply:
Changed.